cli/asc.js

```diff
@@ -270,6 +271,91 @@ exports.main = function main(argv, options, callback) {
 
   // Set up base directory
   const baseDir = args.baseDir ? path.resolve(args.baseDir) : ".";
+  const target = args.target;
+
+  // Once the baseDir is calculated, we can resolve the config, and its extensions
+  let asconfig = getAsconfig(args.config, baseDir, readFile);
+  let asconfigDir = baseDir;
+
+  const seenAsconfig = new Set();
+  seenAsconfig.add(path.join(baseDir, args.config));
+
+  while (asconfig) {
+    // merge target first, then merge options, then merge extended asconfigs
+    if (asconfig.targets && asconfig.targets[target]) {
+      args = optionsUtil.merge(exports.options, asconfig.targets[target], args);
+    }
+    if (asconfig.options) {
+      if (asconfig.options.transform) {
+        // ensure that a transform's path is relative to the current config
+        asconfig.options.transform = asconfig.options.transform.map(p => {
+          if (!path.isAbsolute(p)) {
+            if (p.startsWith(".")) {
+              return path.join(asconfigDir, p);
+            }
+            return require.resolve(p);
+          }
+          return p;
+        });
+      }
+      args = optionsUtil.merge(exports.options, args, asconfig.options);
+    }
+
+    // entries are added to the compilation
+    if (asconfig.entries) {
+      for (const entry of asconfig.entries) {
+        argv.push(
+          path.isAbsolute(entry)
+            ? entry
+            // the entry is relative to the asconfig directory
+            : path.join(asconfigDir, entry)
+        );
+      }
+    }
+
+    // asconfig "extends" another config, merging options of it's parent
+    if (asconfig.extends) {
+      asconfigDir = path.isAbsolute(asconfig.extends)
+        // absolute extension path means we know the exact directory and location
+        ? path.dirname(asconfig.extends)
+        // relative means we need to calculate a relative asconfigDir
+        : path.join(asconfigDir, path.dirname(asconfig.extends));
+      const fileName = path.basename(asconfig.extends);
+      const filePath = path.join(asconfigDir, fileName);
+      if (seenAsconfig.has(filePath)) {
+        asconfig = null;
+      } else {
+        seenAsconfig.add(filePath);
+        asconfig = getAsconfig(fileName, asconfigDir, readFile);
+      }
+    } else {
+      asconfig = null; // finished resolving the configuration chain
+    }
+  }
+
+  // If showConfig print args and exit
+  if (args.showConfig) {
+    stderr.write(JSON.stringify(args, null, 2));
+    return callback(null);
+  }
+
+  // This method resolves a path relative to the baseDir instead of process.cwd()
+  function resolve(arg) {
+    if (path.isAbsolute(arg)) return arg;
+    return path.resolve(path.join(baseDir, arg));
```

Comment:
Wondering if this can be simplified to just `path.resolve(baseDir, arg)`? [See](https://nodejs.org/api/path.html#path_path_resolve_paths):

> The given sequence of paths is processed from right to left, with each subsequent path prepended until an absolute path is constructed.

Reply:
I believe I tried this and it caused the tests to fail. I can resubmit another commit to this to try it.

Reply:
What happens here is `baseDir` can be `.` or `./some/folder`. So `baseDir` and `arg` need to be resolved in the process. Let me see.

Reply:
I was definitely mistaken.